runtime/runtime-core/common/src/aws/smithy/kotlin/runtime/smithy/DocumentBuilder.kt

```diff
@@ -7,46 +7,46 @@ package aws.smithy.kotlin.runtime.smithy
 import kotlin.jvm.JvmName
 
 class DocumentBuilder internal constructor() {
-    val content: MutableMap<String, Document> = linkedMapOf()
+    val content: MutableMap<String, Document?> = linkedMapOf()
 
     infix fun String.to(value: Number?) {
-        require(content[this] == null) { "Key $this is already registered in builder" }
-        content[this] = if (value != null) Document(value) else Document.Null
+        require(!content.containsKey(this)) { "Key $this is already registered in builder" }
```

Comment:
**Style**: Kotlin provides the `in` and `!in` operators for `contains` checks. This might be more idiomatic as:

```kotlin
require(this !in content) { "..." }
```